src/WebJobs.Script.WebHost/DependencyInjection/JobHostScopedServiceProviderFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.DependencyInjection
+{
+    /// <summary>
+    /// An <see cref="IServiceProviderFactory{TContainerBuilder}"/> implementation that creates
+    /// and populates an <see cref="JobHostServiceProvider"/> that can be used as the <see cref="IServiceProvider"/>
+    /// </summary>
+    public class JobHostScopedServiceProviderFactory : IServiceProviderFactory<IServiceCollection>
```

Comment:
Should and associated types be named ScriptHost*? Would be good to ensure our naming is consistent as we add all these new types. In my mind, all of the ScriptHost specific functionality should have that prefix (as we have elsewhere in this review), and if it is truly JobHost specific, it would belong in the core SDK?